### ISSUE TITLE
SBND PSD Map has the opposite numbering convention 

### ISSUE THREAD

**Issue by ikatza** [SBNSoftware/sbndcode#12]
The convention used for the optical detector channel numbers is effectively opposite to the numbering scheme of TPCs. 

Effectively the convention used for channels numbers, and other identifiers is:

> **EVEN** numbers are given to channels on the APA with x>0, and
 **ODD** numbers given to channels on the APA with x<0.

For clarity this convention is shown in the picture below:
![Screenshot 2020-07-21 at 10 58 39 PM](https://user-images.githubusercontent.com/8713617/88111660-399fb900-cba6-11ea-90bd-0feb44e36dfa.png)

For the benefit of a user to select collections of optical detector, I created other indices in the map:

- `pds_box`: to identify all the opdets that are fixated to the same optical module
- `tpc`: to identify all the opdets that belong to one TPC

This last identifier is opposite to the convention sbndcode currently uses for the TPCs:
- tpc0: has x<0
- tpc1: has x>0

These contradictory conventions don't hinder the simulation and as long as everyone who uses them is aware of this asymmetry there shouldn't be any issue.

I believe the PDS map channels, nor any other identifier there, has to change to fix this issue. It would be enough to update the spreadsheet to swap convention.

I'm not knowledgable enough about how or where in the code the channel numbers are assigned to fix this myself.

**Comment by ikatza** [comment]
There's an ongoing effort to update the geometry, that's taking place in PR #6 by @mrguzzo.

This geometry update would be a good opportunity to fix this minor issue, particularly because that update and the fix for this issue are both backwards incompatible: files created with previous geometries won't work with the new geometry.

Tagging @ggamezdiego as convener of the optical simulation in SBND.

**Comment by ggamezdiego** [comment]
If I'm not mistaken, the TPC-ID is an information you can get from the geometry services. I would suggest (unless not possible for some reason) to use this service to get this information in a coherent manner.

**Comment by ikatza** [comment]
Correct, one can get the TPC-ID from geometry services. However, I failed to understand how this would address the rest of IDs.

**Comment by ggamezdiego** [comment]
If with the "rest of IDs" you mean optical channel ones, you are right, they are not available by the geometry services. We are planning to include these possibility in a (near)future update.

**Comment by ikatza** [comment]
I have traced the reason for this, and it essentially comes down to:

- The current channel numbering convention is :         
**the channel number increases with decreasing Z, Y or X; in that order of priority**
- IMO the channel numbering convention should be:  
**the channel number increases with increasing  Z, Y or X; in that order of priority**

For the case of SBND I believe the best is to change the convention, this way it will integrate better with utils and will become more intuitive. The effort in doing that is not huge and I believe will save people a headache in the long run.

This change will render previously produced files incompatible; like simulated events and photon libraries. Again PR#6 will have the same effect so I believe this is a good time to make both changes. 

I have created PR #17 to address this. It's not ready yet, it still requires an update to the PDS Map.

**Comment by ggamezdiego** [comment]
If I remember well, the sorting rule of the PDS channels is the same than for any other sorted objects in LArSoft. I mean, LArSoft has one single sorting rule. I don't think changing it for the PDS is a good idea (it is easier to be consistent and coherent). As I have mentioned in previous comments (in other issues) all the PDS identification/location should be managed by a geometry service where the sorting rule would be hidden, and the user, by "geo-pds-methods" would access easily (an intuitively) any information.

**Comment by ikatza** [comment]
There are distinct sorting rules for various things already in LArSoft:
https://github.com/LArSoft/larcorealg/blob/develop/larcorealg/Geometry/GeoObjectSorter.h

This is by design, a set comparisons might not make sense for every system. However most of them follow the general convention that I suggest. The only one that deviates from that is `sortPlaneStandard()` in which the X coordinate goes opposite.

On SBND however, we override some of those sorting rules to accommodate our specific geometry:
https://github.com/SBNSoftware/sbndcode/blob/develop/sbndcode/Geometry/GeoObjectSorterSBND.h

The SBND overridden `CryostatSorter()`, `TPCSorter()`, `PlaneSorter()` follow the general convention that I propose **the channel number should increase with increasing X, Y or Z** (in some combination). Only `WireSorter()` has a more intricate convention, but still mostly follows that convention.

The `sortorderOpDets()`, which we don't override, is the only one that follows the opposite convention:
https://github.com/LArSoft/larcorealg/blob/develop/larcorealg/Geometry/GeoObjectSorter.cxx

I guess this was a typo. It might be more meaningful to open an issue on `larcorealg`, but that will take a longer time to get accepted/approved if at all. Given that it would break the convention used by experiments that are already running.

**Comment by ggamezdiego** [comment]
Probably, at this point, you should present the case to the SBND PAT meeting and see what the people think.

**Comment by ikatza** [comment]
I've opened a new PR #26  to start from the new changes to the geometry and avoid merge conflicts.

Below I show how the new map looks like. Note that in the plots below the target is to the left, i.e. the beam is coming from left to right. On the spreadsheet that I posted above the beam is coming from the opposite direction. 

The sizes of the detectors are not to scale, but their centre position is. 

To be absolutely clear: this PR doesn't change the positions of the detectors. Only assigns them new channel numbers.

With this PR all ids are consistent, and consistent with the ids from the geometry service. As a side note, I tried using the geometry function `geometry->PositionToTPCID(opdet_xyz)`  to get the TPC index of the optical detectors but it didn't work. I guess because the optical detectors are outside the TPCs. 

![SBND_PDSMap_TPC0](https://user-images.githubusercontent.com/8713617/92273934-35c0bf80-eee4-11ea-861e-47e56d801712.png)

![SBND_PDSMap_TPC1](https://user-images.githubusercontent.com/8713617/92273941-378a8300-eee4-11ea-9321-5e8d6e056dac.png)
